pkg/ticker/ticker.go

```diff
@@ -4,25 +4,119 @@ import (
 	"time"
 
 	"github.com/openservicemesh/osm/pkg/announcements"
+	"github.com/openservicemesh/osm/pkg/configurator"
 	"github.com/openservicemesh/osm/pkg/kubernetes/events"
 	"github.com/openservicemesh/osm/pkg/logger"
 )
 
+const (
+	// Value used to identify ticker stop (0)
+	stopTickerDuration = time.Duration(0)
+	// Minimum allowed ticker duration
+	minimumTickerDuration = time.Duration(1 * time.Minute)
+)
+
+// ResyncTicker contains the stop configuration for the ticker routines
+type ResyncTicker struct {
+	stopTickerRoutine chan struct{}
+	stopConfigRoutine chan struct{}
+}
+
 var (
 	log = logger.New("ticker")
+	// Local reference to global ticker
+	rTicker *ResyncTicker = nil
 )
 
 // InitTicker initializes a global ticker that is configured via
 // pubsub, and triggers global proxy updates also through pubsub.
 // Upon this function return, the ticker is guaranteed to be started
 // and ready to receive new events.
-func InitTicker() {
+func InitTicker(c configurator.Configurator) *ResyncTicker {
```

Comment:
As of this PR `ResyncTicker` is not used anywhere?
What's the plan for that?

Reply:
For now it leaves the room open for the caller to shut down ticker using the channels. This is not used by Catalog but is indeed used by tests, otherwise the parallel routines would be left hanging and running.

It's good to have nonetheless